src/lib/util/Config.ts

```diff
@@ -169,7 +184,7 @@ export class Config {
 		this.configContent = JSON.parse(fileContent);
 
 		// TODO remove this logic before GA, as it is only necessary for short term migrations from old format.
-		if (this.configContent['java-home']) {
+		if (!this.configContent['engines'] && this.configContent['javaHome']) {
```

Comment:
We'll treat a Config.json as prior version only when it is missing `engines` property but already has `java-home`.